src/libraries/System.Runtime/tests/System/DoubleTests.cs

```diff
@@ -272,6 +272,11 @@ public static IEnumerable<object[]> Parse_Valid_TestData()
             yield return new object[] { (567.89).ToString(), defaultStyle, null, 567.89 };
             yield return new object[] { (-567.89).ToString(), defaultStyle, null, -567.89 };
             yield return new object[] { "1E23", defaultStyle, null, 1E23 };
+            yield return new object[] { "9007199254740997.0", defaultStyle, invariantFormat, 9007199254740996 };
+            yield return new object[] { "5.005", defaultStyle, invariantFormat, 5.005 };
+            yield return new object[] { "5.050", defaultStyle, invariantFormat, 5.05 };
+            yield return new object[] { "5005.0", defaultStyle, invariantFormat, 5005 };
+            yield return new object[] { "50050.0", defaultStyle, invariantFormat, 50050 };
 
```

Comment:
It would probably be good to add explicit tests for just cases like `50050` and other scenarios that are integers with trailing zeros but no fractional part specified, to help ensure they don't regress.

Reply:
You got other examples? AFAICT I've covered all the interesting cases?

``` 
9007199254740997.0
5.005
5.050
5005.0
50050.0
5005
050050
0.000
0.005
0.0500
```

Reply:
We know there are bugs in the case where there there are trailing fractional zeros.

I was wanting at least one test covering trailing integral zeros, such as `"50050"` (in contrast to `"50050.0"`).

Reply:
Do any of these cases suggest interseting missing test cases in SingleTests.cs or in DecimalTests.cs?

Reply:
It would probably be good to ensure they are mirrored across the floating-point types (decimal, float, double, and half) to ensure they all have consistent coverage.

Reply:
> Do any of these cases suggest interseting missing test cases in SingleTests.cs or in DecimalTests.cs?

I don't think so. The test cases here test the `TryParseNumber` method. All the other floating point types call `TryParseNumber`, so we should in theory be getting good coverage. I'll update the test cases though, just in case

Reply:
The main difference ends up being in edge case handling.

For example, `double` limits its buffer size to 768 digits, while `single` limits it to 113 digits, and only 20 for half.

It would probably be to validate some number that is, for example, that many digits + a non-zero digit to ensure it is handled correctly.

Imagine, for example, `0.500....0001` which is distinct from `0.500....0000`

Reply:
(I'm not sure if the above case would actually be impacted by the rounding; just giving an example of where we may want to find and validate such a case)

Reply:
Alright I fixed a bug in my changes around rounding when the last digit was == maxDigitCount and last digit > maxDigitCount. I've added some unit tests to cover it. This PR should be good now.